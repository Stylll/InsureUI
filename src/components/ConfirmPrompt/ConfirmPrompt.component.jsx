import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button/Button.component';
import './ConfirmPrompt.styles.scss';

const ConfirmPrompt = ({
  dataTestId,
  message,
  handleCancel,
  handleClick,
  isLoading,
}) => {
  return (
    <div className="confirm-container">
      <div className="confirm-content">
        <div>
          <p className="confirm-title">{message}</p>
        </div>
        <div className="button-container">
        <Button
            title="yes"
            handleClick={handleClick}
            showLoader={isLoading}
            disabled={isLoading}
            dataTestId={`${dataTestId}-yes`}
            customClassName="deleteButton"
        />
        <Button
            title="no"
            handleClick={handleCancel}
            disabled={isLoading}
            dataTestId={`${dataTestId}-no`}
        />
        </div>
      </div>
    </div>
  );
};

ConfirmPrompt.propTypes = {
  handleCancel: PropTypes.func.isRequired,
  handleClick: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
  message: PropTypes.string,
  dataTestId: PropTypes.string,
};

ConfirmPrompt.defaultProps = {
  dataTestId: 'confirmprompt',
  message: 'Are you sure you want to perform this operation?',
};

export default ConfirmPrompt;
