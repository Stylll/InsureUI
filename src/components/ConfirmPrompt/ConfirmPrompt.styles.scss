@import '../../partials/colors';

.confirm-container {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  height: 100%;
  .confirm-title {
    text-align: center;
  }
  .confirm-content {
    box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
    margin: 40px;
    padding: 40px;
    background-color: $white;
  }
  .button-container {
    display: flex;
  }
  .deleteButton {
    background-color: $red;
    color: $white;
  }
}
