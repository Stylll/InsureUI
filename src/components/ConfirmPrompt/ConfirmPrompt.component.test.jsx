import { render, fireEvent } from '@testing-library/react';
import React from 'react';

import ConfirmPrompt from '../../components/ConfirmPrompt/ConfirmPrompt.component';

describe('ConfirmPrompt Test', () => {
  it('should render properly', () => {
    const { baseElement } = render(<ConfirmPrompt
      handleCancel={() => {}}
      handleClick={() => {}}
      isLoading={false}
    />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should call handleCancel function when the no button is clicked', () => {
    const handleCancel = jest.fn();
    const { baseElement, getByTestId } = render(<ConfirmPrompt
      handleCancel={handleCancel}
      handleClick={() => {}}
      isLoading={false}
    />);

    fireEvent.click(getByTestId('confirmprompt-no'));
    expect(handleCancel).toHaveBeenCalled();
  });

  it('should call handleClick function when the yes button is clicked', () => {
    const handleClick = jest.fn();
    const { baseElement, getByTestId } = render(<ConfirmPrompt
      handleCancel={() => {}}
      handleClick={handleClick}
      isLoading={false}
    />);

    fireEvent.click(getByTestId('confirmprompt-yes'));
    expect(handleClick).toHaveBeenCalled();
  });

  it('should display custom message if passed', () => {
    const { getByText } = render(<ConfirmPrompt
      handleCancel={() => {}}
      handleClick={() => {}}
      isLoading={false}
      message="Please do not delete me"
    />);
    expect(getByText('Please do not delete me')).toBeInTheDocument();
  });
});
