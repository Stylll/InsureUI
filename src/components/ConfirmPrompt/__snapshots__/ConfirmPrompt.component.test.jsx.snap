// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmPrompt Test should render properly 1`] = `
<body>
  <div>
    <div
      class="confirm-container"
    >
      <div
        class="confirm-content"
      >
        <div>
          <p
            class="confirm-title"
          >
            Are you sure you want to perform this operation?
          </p>
        </div>
        <div
          class="button-container"
        >
          <div
            class="btn-button"
          >
            <button
              class="button deleteButton"
              data-testid="confirmprompt-yes"
              type="submit"
            >
              yes
            </button>
          </div>
          <div
            class="btn-button"
          >
            <button
              class="button "
              data-testid="confirmprompt-no"
              type="submit"
            >
              no
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
